notebooks/MNIST-torchvision.ipynb

```diff
@@ -0,0 +1,798 @@
+    "        self.conv1 = nn.Conv2d(1, 32, kernel_size=3)\n",
+    "        self.conv2 = nn.Conv2d(32, 64, kernel_size=3)\n",
+    "        self.conv2_drop = nn.Dropout2d(p=dropout)\n",
+    "        self.fc1 = nn.Linear(1600, 100) # 1600 = number channels * width * height\n",
+    "        self.fc2 = nn.Linear(100, 10)\n",
+    "        self.fc1_drop = nn.Dropout(p=dropout)\n",
+    "\n",
+    "    def forward(self, x):\n",
+    "        x = torch.relu(F.max_pool2d(self.conv1(x), 2))\n",
+    "        x = torch.relu(F.max_pool2d(self.conv2_drop(self.conv2(x)), 2))\n",
+    "        \n",
+    "        # flatten over channel, height and width = 1600\n",
+    "        x = x.view(-1, x.size(1) * x.size(2) * x.size(3))\n",
+    "        \n",
+    "        x = torch.relu(self.fc1_drop(self.fc1(x)))\n",
+    "        x = torch.softmax(self.fc2(x), dim=-1)\n",
+    "        return x"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 115,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "torch.manual_seed(0)\n",
+    "\n",
+    "cnn = NeuralNetClassifier(\n",
+    "    Cnn,\n",
+    "    max_epochs=10,\n",
+    "    lr=0.0002,\n",
+    "    optimizer=torch.optim.Adam,\n",
+    "    device=device,\n",
+    "    iterator_train__num_workers=4,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 116,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "  epoch    train_loss    valid_acc    valid_loss     dur\n",
+      "-------  ------------  -----------  ------------  ------\n",
+      "      1        \u001b[36m0.9299\u001b[0m       \u001b[32m0.9307\u001b[0m        \u001b[35m0.2432\u001b[0m  6.9704\n",
+      "      2        \u001b[36m0.3104\u001b[0m       \u001b[32m0.9539\u001b[0m        \u001b[35m0.1505\u001b[0m  7.0305\n",
+      "      3        \u001b[36m0.2220\u001b[0m       \u001b[32m0.9664\u001b[0m        \u001b[35m0.1168\u001b[0m  7.0733\n",
+      "      4        \u001b[36m0.1784\u001b[0m       \u001b[32m0.9713\u001b[0m        \u001b[35m0.0988\u001b[0m  7.0218\n",
+      "      5        \u001b[36m0.1550\u001b[0m       \u001b[32m0.9745\u001b[0m        \u001b[35m0.0882\u001b[0m  6.8893\n",
+      "      6        \u001b[36m0.1426\u001b[0m       \u001b[32m0.9753\u001b[0m        \u001b[35m0.0800\u001b[0m  6.8398\n",
+      "      7        \u001b[36m0.1304\u001b[0m       \u001b[32m0.9775\u001b[0m        \u001b[35m0.0731\u001b[0m  7.0749\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "Exception ignored in: <bound method Image.__del__ of <PIL.Image.Image image mode= size=0x0 at 0x7F2D7F86AFD0>>\n",
+      "Traceback (most recent call last):\n",
+      "  File \"/home/marian/anaconda3/envs/skorch/lib/python3.6/site-packages/PIL/Image.py\", line 616, in __del__\n",
+      "    if (hasattr(self, 'fp') and hasattr(self, '_exclusive_fp')\n",
+      "KeyboardInterrupt\n"
```

Comment:
For the final notebook, let the model train to the end or set the number of epochs to 7.